### PR TITLE
Fix various misspellings and linter items

### PR DESCRIPTION
**Pull request by jsoref**
## Description

This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/knative-client/commit/94cf25e82b095fe1583a395f68ad193a80de3a78#commitcomment-43154244

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/knative-client/commit/a085e32190331d342044106a39ae76d3ddeb5ac3

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

## Changes

* spelling

## Reference

Fixes #

I'm going to delay doing this until I get feedback. It'd be 🐛, but I expect to iterate before I get here.

/lint

### CODE REVIEW

File: .github/pull-request-template.md [hunk 1]
```diff
@@ -17,7 +17,7 @@
 Fixes #
 
 <!--
-Please add an entrty to CHANGELOG.adoc file, too, as part of your Pull Request.
+Please add an entry to CHANGELOG.adoc file, too, as part of your Pull Request.
```
Comment (jsoref) [inline .github/pull-request-template.md:20]: I know there's some irony that I haven't followed this instruction yet. But I will once the rest of the changes are reviewed.

File: pkg/util/unstructured.go [hunk 2]
```diff
@@ -32,8 +32,8 @@ func ToUnstructuredList(obj runtime.Object) (*unstructured.UnstructuredList, err
 		if err != nil {
 			return nil, err
 		}
-		for _, obji := range items {
-			ud, err := toUnstructured(obji)
+		for _, item := range items {
+			ud, err := toUnstructured(item)
```
Comment (jsoref) [inline pkg/util/unstructured.go:36]: This is a more opinionated change. I understand there's already an `obj` in scope, but there's also an `items` and normally while iterating over a list of `foos` the thing one gets is a `foo`, so `item` seems like a pretty reasonable choice.

File: pkg/wait/test_wait_helper.go [hunk 3]
```diff
@@ -42,7 +42,7 @@ func NewFakeWatch(events []watch.Event) *FakeWatch {
 	}
 }
 
-// Stop the watch challend
+// Stop the watch channel called
```
Comment (jsoref) [inline pkg/wait/test_wait_helper.go:45]: I have absolutely no idea what the right item is here.
Things I considered:
`channel`, `called`, `channel called`, `called count`

This was chosen after the equivalent of a coin toss. I asked some others for opinions and ...

**Review by jsoref — COMMENTED** [review]
Most corrections were selected automatically by Google Sheets. There are a couple of items where I made my own fixes. All fault is mine. Please accept my apologies if I've corrupted anything.

File: pkg/wait/test_wait_helper.go (hunk 3 above)
Comment (navidshaikh) [reply pkg/wait/test_wait_helper.go:45]: `Stop the watch channel`

**Review by navidshaikh — APPROVED** [review]
/ok-to-test

Thank you @jsoref for contribution! Can you please squash the commits ?

File: cmd/kn/main_test.go [hunk 4]
```diff
@@ -142,9 +142,9 @@ func TestUnknownCommands(t *testing.T) {
 		expectedError []string
 	}{
 		{
-			[]string{"service", "udpate", "test", "--scale-min=0"},
+			[]string{"service", "update", "test", "--scale-min=0"},
 			[]string{"service"},
-			[]string{"unknown sub-command", "udpate"},
+			[]string{"unknown sub-command", "update"},
```
Comment (navidshaikh) [inline cmd/kn/main_test.go:147]: this test expects an error given an uknown command, revert this change to make unit tests pass

File: pkg/kn/config/config.go [hunk 5]
```diff
@@ -97,7 +97,7 @@ var globalConfig = config{}
 // GlobalConfig is the global configuration available for every sub-command
 var GlobalConfig Config = &globalConfig
 
-// bootstrapConfig reads in config file and boostrap options if set.
+// bootstrapConfig reads in config file and bootstrap options if set.
```
Comment (daisy-ycguo) [inline pkg/kn/config/config.go:100]: `bootstrapConfig` in the comment line should be `BootstrapConfig`, beginning with a capital letter.

File: pkg/util/unstructured.go (hunk 2 above)
Comment (daisy-ycguo) [reply pkg/util/unstructured.go:36]: I'm fine with `item`. Thank you.

**Review by daisy-ycguo — CHANGES_REQUESTED** [review]
Thank you for this PR. I just have a mirror comment.

File: cmd/kn/main_test.go (hunk 4 above)
Comment (jsoref) [reply cmd/kn/main_test.go:147]: Ah, would `unknown` be acceptable instead?

File: pkg/serving/v1/client_mock.go [hunk 6]
```diff
@@ -181,7 +181,7 @@ func (sr *ServingRecorder) GetConfiguration(name string, config *servingv1.Confi
 
 }
 
-// Check for the base reviision
+// Check for the base revision
```
[the remark this answers is not in the document]
Comment (jsoref) [reply pkg/serving/v1/client_mock.go:184]: If someone wants, I could copy a suggestion, but, there are other items nearby which would also fail this style check, it seems like it'd be better if someone familiar w/ the file/project fixed the whole file. (I'm bumbling through as can be seen by the rest of this PR.)

**Comment by jsoref** [comment]
Thanks for the feedback, I think I've addressed the items. (Sorry about the multiple forced-pushes, I'm having trouble reading GitHub today...)

I think this is close to ready. What should my steps going forward be?

Some projects tend ask me to collapse PRs like this into a single commit. Is that desired by this project?

5e3f4d7f17c0b2079c77a7d032f2857ab876d8ec seems to be a sample commit for cleanup, so I suspect that the PR should be a single commit titled `chore: spelling (#1057)` w/o an entry in the `CHANGELOG.adoc` file.

Collapsing / picking a commit message would be the last thing I'll do, once everyone is happy with the rest of the PR.

**Review by maximilien — APPROVED** [review]
Wonderful PR. Please add changes to Changelog so we can LGTM this. Cheers 🍻

File: cmd/kn/main_test.go (hunk 4 above)
Comment (navidshaikh) [reply cmd/kn/main_test.go]: sure

**Review by navidshaikh — APPROVED** [review]
/lgtm
/approve

Thank you @jsoref ! 
There are other github actions configured that are run against PRs, for example reviewdog etc, I think we can include this spell checker as well. cc: @mattmoor

**Comment by mattmoor** [comment]
This is what we run now: https://github.com/knative-sandbox/.github/blob/628760d28ae8b4f3edf2ed3478586d81cbbfd578/workflow-templates/knative-style.yaml#L124-L145
